pkg/operator/observe_config.go

```diff
@@ -0,0 +1,167 @@
+	queue workqueue.RateLimitingInterface
+
+	rateLimiter flowcontrol.RateLimiter
+	observers   []observeConfigFunc
+}
+
+func NewConfigObserver(
+	operatorConfigInformer operatorconfiginformerv1alpha1.KubeControllerManagerOperatorConfigInformer,
+	kubeInformersForOpenshiftServiceCertSignerNamespace informers.SharedInformerFactory,
+	operatorConfigClient operatorconfigclientv1alpha1.KubecontrollermanagerV1alpha1Interface,
+	kubeClient kubernetes.Interface,
+	clientConfig *rest.Config,
+) *ConfigObserver {
+	c := &ConfigObserver{
+		operatorConfigClient: operatorConfigClient,
+		kubeClient:           kubeClient,
+		clientConfig:         clientConfig,
+
+		queue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "ConfigObserver"),
+
+		rateLimiter: flowcontrol.NewTokenBucketRateLimiter(0.05 /*3 per minute*/, 4),
+		observers:   []observeConfigFunc{observeCABundleConfig},
+	}
+
+	operatorConfigInformer.Informer().AddEventHandler(c.eventHandler())
+	kubeInformersForOpenshiftServiceCertSignerNamespace.Core().V1().ConfigMaps().Informer().AddEventHandler(c.eventHandler())
+
+	return c
+}
+
+// sync reacts to a change in controller manager images.
+func (c ConfigObserver) sync() error {
+	var err error
+	observedConfig := map[string]interface{}{}
+
+	for _, observer := range c.observers {
+		observedConfig, err = observer(c.kubeClient, c.clientConfig, observedConfig)
+		if err != nil {
+			return err
+		}
+	}
+
+	operatorConfig, err := c.operatorConfigClient.KubeControllerManagerOperatorConfigs().Get("instance", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	// don't worry about errors
+	currentConfig := map[string]interface{}{}
+	json.NewDecoder(bytes.NewBuffer(operatorConfig.Spec.ObservedConfig.Raw)).Decode(&currentConfig)
+	if reflect.DeepEqual(currentConfig, observedConfig) {
+		return nil
+	}
+
+	glog.Infof("writing updated observedConfig: %v", diff.ObjectDiff(operatorConfig.Spec.ObservedConfig.Object, observedConfig))
+	operatorConfig.Spec.ObservedConfig = runtime.RawExtension{Object: &unstructured.Unstructured{Object: observedConfig}}
+	if _, err := c.operatorConfigClient.Update(operatorConfig); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func observeCABundleConfig(kubeClient kubernetes.Interface, clientConfig *rest.Config, observedConfig map[string]interface{}) (map[string]interface{}, error) {
+	// sync our target configmap locally
+	caBundleConfig, _, err := resourceapply.SyncConfigMap(kubeClient, serviceCertSignerNamespaceName, "signing-cabundle", targetNamespaceName, "signing-cabundle")
+	if errors.IsNotFound(err) {
+		return nil, fmt.Errorf("unable to find requred ConfigMap containing signing CA bundle info: %v", err)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	if val := caBundleConfig.Data["cabundle.crt"]; len(val) > 0 {
+		// TODO: what is the nested field path? Where do we store the service-ca info?
```

Comment:
@deads2k Once we've copied the ca-bundle configmap to our namespace, and verified that it has the value we want, where in the observedConfig should we store that value?